server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -129,6 +127,12 @@ public FastenKafkaPlugin(boolean enableKafka, Properties consumerProperties, Pro
         logger.debug("Constructed a Kafka plugin for " + plugin.getClass().getCanonicalName());
     }
 
+    private Properties cloneKafkaPropWithNewClientId(Properties consumerProperties, String suffix) {
+        Properties consumerPrioProperties = (Properties) SerializationUtils.clone(consumerProperties);
```

Comment:
I did not realize this before... Why do we ned to de-/serialize the whole properties object for cloning? This introduces quite a bit of overhead. If Kafka indeed does have no way of simply copying the options, I would propose that we try move the properties creation into a static helper method that can be used from both the `FastenServer` and from this location.

Reply:
AFAIK, Kafka doesn't have a way to copy properties.
I have addressed this in d69255b by creating a static method.